qiskit/primitives/base/base_primitive.py

```diff
@@ -72,3 +73,23 @@ def _cross_validate_circuits_parameter_values(
         return validation._cross_validate_circuits_parameter_values(
             circuits, parameter_values=parameter_values
         )
+
+
+BasePrimitive = BasePrimitiveV1
+
+
+class BasePrimitiveV2(ABC):
+    """Primitive abstract base class version 2."""
+
+    version = 2
+    _options_class: type[BasePrimitiveOptions] = BasePrimitiveOptions
+
+    def __init__(self, options: BasePrimitiveOptionsLike | None = None):
+        self._options = self._options_class()
+        if options:
+            self._options.update(options)
+
+    @property
+    def options(self) -> BasePrimitiveOptions:
```

Comment:
There seems to be no setter of options. How can users specify the number of shots for each run? Do they need to make a new Estimator object with different options?

Reply:
The only way to do it is to set the attribute of the option like:

```python
estimator = Estimator()
estimator.run(pubs)
estimator.options.execution.shots = 1024 # new shots
estimator.run(pubs)
```